server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -85,18 +145,62 @@ public String[] excludes() {
         return this.excludes;
     }
 
-    public boolean hasFilter() {
-        return this.includes.length > 0 || this.excludes.length > 0;
+    public Boolean includeVectors() {
+        return this.includeVectors;
     }
 
     public SourceFilter filter() {
-        return new SourceFilter(includes, excludes);
+        return filter(null);
+    }
+
+    public SourceFilter filter(MappingLookup mappingLookup) {
+        if (includeVectors != null && includeVectors) {
```

Comment:
My concerns here are:

 - Whats the behavior if `_include_vectors` is true, but an individual vector field is specifically excluded?
 - What if `_include_vectors` is false, but an individual vector field is specifically included?

This is tricky as setting `_include_vectors` to `false`, but then specifically including a vector field seems different than specifically excluding and including a particular field. 

Maybe it should be treated the same as `exclude: foo*, include: foobar`.